packages/camera/android/src/main/java/io/flutter/camera/CameraPlugin.java

```diff
@@ -518,37 +529,30 @@ public void notImplemented() {}
           });
     }
 
+    private void writeToFile(ByteBuffer buffer, File file) throws IOException {
+      try (FileOutputStream outputStream = new FileOutputStream(file)) {
+        while (0 < buffer.remaining()) {
+          outputStream.getChannel().write(buffer);
+        }
+      }
+    }
+
     void capture(String path, final Result result) {
       final File file = new File(path);
       imageReader.setOnImageAvailableListener(
           new ImageReader.OnImageAvailableListener() {
             @Override
             public void onImageAvailable(ImageReader reader) {
-              Image image = null;
-              try {
-                image = reader.acquireLatestImage();
+              boolean success = false;
+              try (Image image = reader.acquireLatestImage()) {
                 ByteBuffer buffer = image.getPlanes()[0].getBuffer();
-                byte[] bytes = new byte[buffer.capacity()];
-                buffer.get(bytes);
-                OutputStream output = null;
-                try {
-                  output = new FileOutputStream(file);
-                  output.write(bytes);
-                  result.success(null);
-                } finally {
-                  if (output != null) {
-                    output.close();
-                    if (started) {
-                      start();
-                    }
-                  }
-                }
+                writeToFile(buffer, file);
+                success = true;
               } catch (IOException e) {
                 result.error("IOError", "Failed saving image", null);
-              } finally {
-                if (image != null) {
-                  image.close();
-                }
+              }
+              if (success) {
+                result.success(null);
```

Comment:
I think we should be able to dispense with the `success` flag by moving the `result.success` call to just after `writeToFile`.

Reply:
The code is wrong, but in case image.close() throws I don't want to double report.
Updated